crates/sui-source-validation-service/src/lib.rs

```diff
@@ -579,3 +587,32 @@ async fn check_version_header<B>(
 async fn list_route(State(_app_state): State<Arc<RwLock<AppState>>>) -> impl IntoResponse {
     (StatusCode::OK, "").into_response()
 }
+
+pub fn start_prometheus_server(addr: TcpListener) -> RegistryService {
```

Comment:
This inspired by [`sui-proxy`](https://sourcegraph.com/github.com/MystenLabs/sui/-/blob/crates/sui-proxy/src/metrics.rs?L17:8-17:31). We have other parts in the code with `start_prometheus_server` that roughly follow the same structure.